Cargo.toml

```diff
@@ -56,6 +56,8 @@ export_derive = ["logos-derive"]
 std = []
 # Use safe alternatives for unsafe code (may impact performance)?
 forbid_unsafe = ["logos-derive?/forbid_unsafe"]
+# Use the state machine codegen instead of the tailcall codegen
+state_machine_codegen = ["logos-derive?/state_machine_codegen"]
```

Comment:
I think this feature should be heavily documented. E.g., if one enables/disables this, will it change the behavior of Logos for some regexes (i.e., does this feature add/remove support for some regex patterns)?

Reply:
Added a note in the book.